pkg/server/server.go

```diff
@@ -125,6 +125,9 @@ type Server struct {
 	rwlock  sync.RWMutex
 	clients map[uint64]*clientConn
 
+	userResLock  sync.RWMutex // userResLock used to protect userResource
```

Comment:
Please request review from @tiancaiamao to estimate the performance for 2M user's case